project1-sps/script.js

```diff
@@ -1,4 +1,79 @@
+var NumPlayerWins = 0;
+var NumPlayerlosses = 0;
+var NumPlayerdraws = 0;
+
+var mode = "get user name";
+var UserName = "";
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue;
+  if (mode == "get user name") {
+    //save username in global variable
+    UserName = input;
+    // change mode to play game
+    mode = "play game";
+    myOutputValue = `Hello ${UserName}`;
+  } else {
+    var playerInput = input;
+    var computerInput = getRandomObject();
+    console.log(computerInput);
+
+    var allWinningConditions =
+      (playerInput == SCISSORS && computerInput == PAPER) ||
+      (playerInput == STONE && computerInput == SCISSORS) ||
+      (playerInput == PAPER && computerInput == STONE);
+
+    var allLosingConditions =
+      (computerInput == SCISSORS && playerInput == PAPER) ||
+      (computerInput == STONE && playerInput == SCISSORS) ||
+      (computerInput == PAPER && playerInput == STONE);
```

Comment:
Good way of storing booleans, these will make your if statements much more concise and readable!